module/enrichers.mjs

```diff
@@ -967,6 +1044,30 @@ async function rollAction(event) {
     try {
       switch ( type ) {
         case "damage": return await rollDamage(event);
+        case "item":
+          if ( target.dataset.rollItemUuid ) {
+            return (await fromUuid(target.dataset.rollItemUuid)).use();
+          } else if ( target.dataset.rollItemName ) {
+            if ( target.dataset.rollItemActor ) {
+              const gameActor = await fromUuid(target.dataset.rollItemActor);
+              if ( gameActor.testUserPermission(game.user, "OWNER") ) {
+                if ( canvas.tokens.controlled[0]?.actor.items.getName(target.dataset.rollItemName) ) {
+                  return canvas.tokens.controlled[0]?.actor.items.getName(target.dataset.rollItemName).use();
+                } else if ( gameActor.items.getName(target.dataset.rollItemName) ) {
+                  await gameActor.items.getName(target.dataset.rollItemName).use();
+                  if ( canvas.tokens.controlled.length > 0 ) {
+                    ui.notifications.warn(`Your controlled token ${canvas.tokens.controlled[0].name} does not have an Item with name ${target.dataset.rollItemName} using Item Enricher owner`);
+                  }
+                  return;
+                } else {
+                  return ui.notifications.warn(`${gameActor.name} does not have an Item with name ${target.dataset.rollItemName}.`);
+                }
+              }
+              return dnd5e.documents.macro.rollItem(target.dataset.rollItemName);
+            } else {
+              return dnd5e.documents.macro.rollItem(target.dataset.rollItemName);
+            }
```

Comment:
Okay, I've gone over this logic to make the flow a bit clearer, add some comments, and remove some levels of indentation. The functionality should be the same, but let me know if you notice any issues.

```suggestion
          let item;
          // If UUID is provided, always roll that item directly
          if ( target.dataset.rollItemUuid ) item = await fromUuid(target.dataset.rollItemUuid);
          else if ( target.dataset.rollItemName ) {
            const actor = target.dataset.rollItemActor ? await fromUuid(target.dataset.rollItemActor) : null;

            // If no actor is specified or player isn't owner, fall back to the macro rolling logic
            if ( !actor?.isOwner ) return rollItem(target.dataset.rollItemName);

            // If a token is controlled and it has an item with the correct name, activate it
            if ( canvas.tokens.controlled[0]?.actor.items.getName(target.dataset.rollItemName) ) {
              item = canvas.tokens.controlled[0]?.actor.items.getName(target.dataset.rollItemName);
            }

            // Otherwise check the specified actor for the item
            else {
              item = actor.items.getName(target.dataset.rollItemName);

              // Display a warning to indicate the item wasn't rolled from the controlled actor
              if ( item && (canvas.tokens.controlled.length > 0) ) ui.notifications.warn(
                game.i18n.format("MACRO.5eMissingTargetWarn", {
                  actor: canvas.tokens.controlled[0].name, item: target.dataset.rollItemName
                })
              );
            }

            // If no item could be found at all, display a warning
            if ( !item ) ui.notifications.warn(game.i18n.format("EDITOR.DND5E.Inline.Warning.NoItemOnActor", {
              actor: actor.name, item: target.dataset.rollItemName
            }));
          }
          if ( item ) item.use();
          return;
```

This also requires adjusting in a few other places:

1) At the top of the file, I imported `rollItem` rather than refering to it using the whole path:

```javascript
import { rollItem } from "./documents/macro.mjs";
```

2) I adjusted a localization name for the warning, so the line below the shows `NoActorWarning` should be adjusted:

```javascript
        ui.notifications.warn("EDITOR.DND5E.Inline.Warning.NoActor", { localize: true });
```

3) New localization entries should be added to `EDITOR.DND5E.Inline`:

```json
  "Warning": {
    "NoActor": "No selected or assigned actor could be found to execute this roll.",
    "NoItemOnActor": "{actor} does not have an Item with name {item}.",
  }
```

And the "NoActorWarning" line should be removed.

Reply:
Looks good to me, updated. 
Thank you Jeff, I appreciate it